opa-manager-operator/src/charm.py

```diff
@@ -0,0 +1,220 @@
+            "audit_cli_args": self._audit_cli_args(),
+            "crs": crs,
+            "namespace": os.environ["JUJU_MODEL_NAME"],
+        }
+
+        template = self._render_jinja_template(
+            "files/pod-spec.yaml.jinja2", template_args
+        )
+
+        spec = yaml.load(template)
+        return spec
+
+    def _cli_args(self):
+        """
+        Construct command line arguments for OPA
+        """
+
+        args = [
+            "--operation=audit",
+            "--operation=status",
+            "--logtostderr",
+            "--port=8443",
+            "--logtostderr",
+            f"--exempt-namespace={os.environ['JUJU_MODEL_NAME']}",
+            "--operation=webhook",
+        ]
+        return args
+
+    def _audit_cli_args(self):
+        """
+        Construct command line arguments for OPA
+        """
+
+        args = [
+            "--operation=audit",
+            "--operation=status",
+            "--logtostderr",
+        ]
+
+        return args
+
+    def _check_config(self):
+        """
+        Identify missing but required items in configuation
+        :returns: list of missing configuration items (configuration keys)
+        """
+        logger.debug("Checking Config")
+        config = self.model.config
+        missing = []
+
+        if not config.get("imagePath"):
+            missing.append("imagePath")
+
+        return missing
+
+    def _render_jinja_template(self, template, ctx):
+        spec_template = {}
+        with open(template) as fh:
+            spec_template = Template(fh.read())
+
+        return spec_template.render(**ctx)
+
+    def _on_start(self, event):
+        k8s_objects = self._load_yaml_objects(["files/psp.yaml"])
+        k8s_objects.append(
+            yaml.load(
+                self._render_jinja_template(
+                    "files/sync.yaml.jinja2",
+                    {"namespace": os.environ["JUJU_MODEL_NAME"]},
+                )
+            )
+        )
+        log(f"K8s objects: {k8s_objects}")
+        for k8s_object in k8s_objects:
+            utils.create_k8s_object(os.environ["JUJU_MODEL_NAME"], k8s_object)
+
+    def _configure_pod(self):
+        """
+        Setup a new OPA pod specification
+        """
+        logger.debug("Configuring Pod")
+
+        missing_config = self._check_config()
+        if missing_config:
+            logger.error(
+                "Incomplete Configuration : {}. "
+                "Application will be blocked.".format(missing_config)
+            )
+            self.unit.status = BlockedStatus(
+                "Missing configuration: {}".format(missing_config)
+            )
+            return
+
+        if not self.unit.is_leader():
+            self.unit.status = ActiveStatus()
+            return
+
+        self.unit.status = MaintenanceStatus("Setting pod spec.")
+        pod_spec = self._build_pod_spec()
+
+        self.model.pod.set_spec(pod_spec)
+        self.app.status = ActiveStatus()
```

Comment:
I think you don't need to set app status here.

Reply:
will remove